File: leetcode-submissions/2958_Length_of_Longest_Subarray_With_at_Most_K_Frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
      unordered_map<int,int>mp;
     int start=0;
     int end=0;
     int cnt=INT_MIN;
     for(;end<nums.size();end++){
        mp[nums[end]]++;
        while(mp[nums[end]]>k){
            mp[nums[start++]]--;
        }
        cnt=max(cnt,end-start+1);

     }
return cnt;
    }
};

File: leetcode-submissions/2958_Length_of_Longest_Subarray_With_at_Most_K_Frequency.ts
function maxSubarrayLength(nums: number[], k: number): number {
    let i :number = 0 , j:number = 0 , maxAns :number = 0;
    const n  : number = nums.length;
    const mp:Map<number , number > = new Map();

    while(j<n){
        mp.set(nums[j], (mp.get(nums[j]) || 0) + 1);
        while(mp.get(nums[j])>k){
            mp.set(nums[i], (mp.get(nums[i]) ||  0 )-1);
            i++;
        }

        maxAns = Math.max(maxAns, j-i+1);
        j++;

    }
    return maxAns;
};